src/cogent3/app/evo.py

```diff
@@ -237,13 +221,12 @@ def _fit_aln(
 
         return lf
 
-    def main(self, aln, initialise=None, construct=True, **opt_args):
+    T = Union[SerialisableType, AlignedSeqsType]
+
+    def main(self, aln: T, initialise=None, construct=True, **opt_args) -> T:
```

Comment:
I've been thinking about this a bit more. For every cogent3 decorated app EXCEPT writers, the input argument should not include `SerialisableType`. The reason being, they really do require a more specific type for execution. Whether an object can be serialised only matters for writers.

In this specific instance, change to
```python
def main(self, aln: AlignedSeqsType, initialise=None, construct=True, **opt_args) -> Union[SerialisableType, AlignedSeqsType]:
```

Reply:
Do I need to remove `SerialisableType` from input of all main functions? and add a condition in decorator to check it?